assets/js/src/index.js

```diff
@@ -0,0 +1,90 @@
+import { tracker } from './tracker';
+
+import {
+	trackBeginCheckout,
+	trackShippingTier,
+	trackListProducts,
+	trackAddToCart,
+	// trackChangeCartItemQuantity,
+	trackRemoveCartItem,
+	trackSelectContent,
+	trackSearch,
+	trackViewItem,
+	trackException,
+} from './tracker/data-formatting';
+
+/**
+ * Register all Google Analytics 4 events that can be tracked
+ */
+tracker.setupEvents( [
+	{
+		/**
+		 * @see https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtag#begin_checkout
+		 */
+		name: 'begin_checkout',
+		callback: trackBeginCheckout,
```

Comment:
:nail_care: 
I got a feeling we're adding more work for ourselves by translating it over and over.
1. `./tracker/data-formatting.js` defines a set of functions `track*`
2. here we map event name like "begin_checkout" to "trackBeginCheckout"
3. `./tracker/index.js` defines a class and implements methods to retrieve `track*` for given name.
4. `integrations/blocks.js` maps block events to our event names
5. `includes/class-wc-google-gtag-js.php` maps WC hook names to our event names

We have 2 disjointed sets of names given by the platform
- WC hooks
- black events
So we introduce our set of `event_names` to handle them. I don't see a reason to introduce fourth set of `trackFunctionNames`.

What I'd propose is to do something like
```js
// tracker/index.js
import * as eventDataFormats from './data-formatting.js';
// …
attach( eventName ) {
  return (data) => this.track( 'event', eventName, eventDataFormats[ eventName ]( data ) );
}
```

and `data-formatting.js` will be the only source of our event names and respective formatters